Luke/BareMetal.py

```diff
@@ -12,21 +13,75 @@
 #    along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 from json import loads
+from logging import getLogger
+
+from .utils.Utils import locate_mac_in_log
+
+logger = getLogger(__name__)
 
 
 class BareMetal(object):
-    def __init__(self, bare_metal_str):
-        self.bare_metal = bare_metal_str
+    def __init__(self, bare_metal_str, hostname=None, ip=None, mac=None):
+        # load everything from the string
         json_bare = loads(bare_metal_str)
         for key, value in json_bare.iteritems():
             self.__dict__[key] = value
-        self.ip = json_bare['ip'] if 'ip' in json_bare else None
+
+        self.ip = ip if ip else self._init_ip(json_bare=json_bare)
+        self.mac = mac if mac else self._init_mac(json_bare=json_bare)
+        self.hostname = hostname if hostname else self._init_hostname(json_bare=json_bare)
+
+    def _init_ip(self, json_bare):
+        # guess ip
+        self.ip = None
```

Comment:
actually it is required, since when I get here self.ip isn't defined yet.
tested